cmd/cache_serve.go

```diff
@@ -145,14 +146,22 @@ func advertiseCache(prefix string, nsAds []director.NamespaceAd) error {
 }
 
 func serveCache( /*cmd*/ *cobra.Command /*args*/, []string) error {
-	defer config.CleanupTempResources()
+	shutdownCtx, shutdownCancel := context.WithCancel(context.Background())
+	var wg sync.WaitGroup
```

Comment:
Can you add a brief comment in one of the `*_serve.go` files that explains how/why you're using the wait group? I think it makes sense after looking up the `sync` package, but it would be nice to have at least one comment in the code that explains the logic.